### ISSUE TITLE
Failure with dev xml2

### ISSUE THREAD

**Issue by hadley** [hugheylab/pmparser#88]
```
══ Failed tests ════════════════════════════════════════════════════════════════
── Failure ('test_modify_pubmed_db_std.R:32:5'): modifyPubmedDb create matches standard ──
journal not equal to DBI::dbReadTable(conExp, tableName).
Attributes: < Component "row.names": Numeric: lengths (0, 32) differ >
Component "pmid": Numeric: lengths (0, 32) differ
Component "journal_name": Lengths (0, 32) differ (string compare on first 0)
Component "journal_iso": Lengths (0, 32) differ (string compare on first 0)
Component "pub_date": Numeric: lengths (0, 32) differ
Component "pub_year": Lengths (0, 32) differ (string compare on first 0)
Component "pub_month": Lengths (0, 32) differ (string compare on first 0)
Component "pub_day": Lengths (0, 32) differ (string compare on first 0)
Component "medline_date": Lengths (0, 32) differ (string compare on first 0)
...
── Failure ('test_modify_pubmed_db_std.R:52:5'): modifyPubmedDb update matches standard ──
journal not equal to DBI::dbReadTable(conExp, tableName).
Attributes: < Component "row.names": Numeric: lengths (32, 55) differ >
Component "pmid": Numeric: lengths (32, 55) differ
Component "journal_name": Lengths (32, 55) differ (string compare on first 32)
Component "journal_iso": Lengths (32, 55) differ (string compare on first 32)
Component "pub_date": Numeric: lengths (32, 55) differ
Component "pub_year": Lengths (32, 55) differ (string compare on first 32)
Component "pub_month": Lengths (32, 55) differ (string compare on first 32)
Component "pub_day": Lengths (32, 55) differ (string compare on first 32)
Component "medline_date": Lengths (32, 55) differ (string compare on first 32)
...
── Error ('test_parse_element.R:35:3'): parseJournal ───────────────────────────
Error in `xml_attr(xml_find_first(x1[idx], "CitedMedium"))`: argument "attr" is missing, with no default
Backtrace:
    ▆
 1. └─pmparser::parseJournal(pmXml, dPmid) at test_parse_element.R:35:3
 2.   ├─data.table::data.table(...)
 3.   └─xml2::xml_attr(xml_find_first(x1[idx], "CitedMedium"))
── Error ('test_parse_element_no_rows.R:14:5'): parseElement returns no rows ───
Error in `xml_attr(xml_find_first(x1[idx], "CitedMedium"))`: argument "attr" is missing, with no default
Backtrace:
    ▆
 1. └─parseFuncs[[i]](pmXml, dPmid) at test_parse_element_no_rows.R:14:5
 2.   ├─data.table::data.table(...)
 3.   └─xml2::xml_attr(xml_find_first(x1[idx], "CitedMedium"))

[ FAIL 4 | WARN 0 | SKIP 4 | PASS 158 ]
```

I'm not sure what's causing the first two failures but the second two are because somehow xml2 previously ignored the fact that you didn't supply an attribute name.

**Comment by hadley** [comment]
Unfortunately I need to submit xml2 by Dec 8 in order to avoid it being pulled from CRAN so my apologies for the short timeline.

**Comment by jakejh** [comment]
Thanks for the heads up. I believe the issue is now fixed in pmparser 1.0.18 03076df, which I just submitted to cran.

**Comment by hadley** [comment]
Great, thanks!
